sdk/go/auto/remote_workspace_test.go

```diff
@@ -111,8 +117,25 @@ func testRemoteStackGitSourceErrors(t *testing.T, fn func(ctx context.Context, s
 	}
 }
 
+// fetchCommitHash runs `git ls-remote URL branch` to determine the latest commit for the given repo
+// URL and branch.
+func fetchCommitHash(url, branch string) (string, error) {
+	cmd := exec.Command("git", "ls-remote", url, branch)
+	output, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	out := strings.TrimSpace(string(output))
+	fields := strings.Fields(out)
+	if len(fields) == 0 {
+		return "", fmt.Errorf("could not determine commit hash from %q", out)
+	}
+	return fields[0], nil
+}
+
 func testRemoteStackGitSource(t *testing.T, fn func(ctx context.Context, stackName string, repo GitRepo,
 	opts ...RemoteWorkspaceOption) (RemoteStack, error),
+	useCommitHash bool,
```

Comment:
I know this is like this from before, but optional readability improvement:
it was confusing to read the signature for fn and the new useCommitHash parameter
because it looked like the line above was the end of the entire function signature.
Consider having it parameter-per-line:

```suggestion

func testRemoteStackGitSource(
	t *testing.T,
	fn func(ctx context.Context, stackName string, repo GitRepo, opts ...RemoteWorkspaceOption) (RemoteStack, error),
	useCommitHash bool,
```

If the line with fn is too long, we can drop the parameter names from the parameter signature;

```
fn func(context.Context, string, GitRepo, ...RemoteWorkspaceOption) (RemoteStack, error),
```